### DISCUSSION TITLE
Table组件rowSelection属性的type值

### DISCUSSION THREAD

**Discussion by lyqyq2001** [ant-design/ant-design#51739] [Q&A]
rowSelection.type只有checkbox和radio这两种类型；当我传入了rowSelection属性时，就一定会有选择框出现，我不希望它出现，例如我可以设置rowSelection.type为undefined使得选择框消失，我应该怎么做

**Comment by petercat-assistant** [comment]
在 Ant Design 的 Table 组件中，如果你不希望选择框出现，可以通过不传入 `rowSelection` 属性来实现。如果你已经传入了 `rowSelection` 属性，但希望在某些条件下不显示选择框，可以考虑以下方法：

1. **条件渲染 `rowSelection` 属性**：在传入 `rowSelection` 属性时，使用条件判断来决定是否传入。例如：

   ```jsx
   const shouldShowSelection = false; // 根据你的条件设置

   <Table
     rowSelection={shouldShowSelection ? { type: 'checkbox' } : undefined}
     columns={columns}
     dataSource={data}
   />
   ```

2. **自定义 `rowSelection` 的渲染**：如果你需要更复杂的逻辑，可以通过自定义 `rowSelection` 的渲染来控制选择框的显示。

如果你需要进一步的帮助或有其他问题，请在评论中描述你的问题并 @petercat-assistant 来与我开始对话。
